Vostok.ClusterClient.Core/IClusterClientConfiguration.cs

```diff
@@ -40,9 +40,15 @@ public interface IClusterClientConfiguration
 
         /// <summary>
         /// <para>An implementation of cluster provider. See <see cref="IClusterProvider.GetCluster"/> for more details.</para>
-        /// <para>This parameter is REQUIRED.</para>
+        /// <para>This parameter or <see cref="AsyncClusterProvider"/> is REQUIRED.</para>
         /// </summary>
         IClusterProvider ClusterProvider { get; set; }
+        
+        /// <summary>
+        /// <para>An implementation of cluster provider. See <see cref="IAsyncClusterProvider.GetClusterAsync"/> for more details.</para>
+        /// <para>This parameter or <see cref="ClusterProvider"/> is REQUIRED.</para>
+        /// </summary>
+        IAsyncClusterProvider AsyncClusterProvider { get; set; }
```

Comment:
Option two: have one `ClusterProvider` field and try to cast it to `IAsyncClusterProvider `.

But I don't like how transformations or calls will work with it.